djautotask/migrations/0092_auto_20210629_1124.py

```diff
@@ -0,0 +1,24 @@
+# Generated by Django 3.1.7 on 2021-06-29 11:24
+
+from django.db import migrations
+
+
+def process_tasks(apps, schema_editor):
+    Task = apps.get_model('djautotask', 'Task')
+
+    # Remove tasks without a project in preparation for migration
+    # removing null=True from project foreign key on Task model.
+    null_project_tasks = Task.objects.filter(project__isnull=True)
+    for task in null_project_tasks:
+        task.delete()
```

Comment:
This actually worries me. In theory its totally fine, but we have that efficiency problem when trying to delete many objects with ranks attached. It's possible this could cause a deadlock during migration or make it run for an hour in a bad case. I'll make up a dummy db with data to mimic that and test it before I merge.

Reply:
Maybe first check if we actually have any customers with a large number of tasks. We may not.

Reply:
Ah, good idea!

Reply:
Yeah good point and thanks for checking that. I was banking on it being quite rare to have a large number of tasks with a null project FK.

Reply:
No clients with Tasks and null projects, should be all good! I'll finish up review and merge shortly.